_stbt/gst_utils.py

```diff
@@ -168,8 +168,27 @@ def __del__(self):
             self._ctx.__exit__(None, None, None)
 
 
+class CapturedFrame(numpy.ndarray):
+    """
+    A subclass of ndarray that we use so we can attach additional metadata to it
+    such as timestamps.
+    """
+    def __new__(cls, array, dtype=None, order=None, _gst_pts=None):
+        obj = numpy.asarray(array, dtype=dtype, order=order).view(cls)
+        obj._gst_pts = _gst_pts  # pylint: disable=protected-access
+        return obj
+
+    def __array_finalize__(self, obj):
+        if obj is None:
+            return
+        self._gst_pts = getattr(obj, '_gst_pts', None)  # pylint: disable=protected-access,attribute-defined-outside-init
```

Comment:
Yep: http://docs.scipy.org/doc/numpy-1.10.1/user/basics.subclassing.html#slightly-more-realistic-example-attribute-added-to-existing-array I'll add it when I rebase.